addons/wiki/models.py

```diff
@@ -82,6 +85,240 @@ def build_wiki_url(node, label, base, end):
     return '/{pid}/wiki/{wname}/'.format(pid=node._id, wname=label)
 
 
+class WikiVersion(ObjectIDMixin, BaseModel):
+    user = models.ForeignKey('osf.OSFUser', null=True, blank=True, on_delete=models.CASCADE)
+    wiki_page = models.ForeignKey('WikiPage', null=True, blank=True, on_delete=models.CASCADE, related_name='versions')
+    content = models.TextField(default='', blank=True)
+    identifier = models.IntegerField(default=1)
+    date = NonNaiveDateTimeField(auto_now_add=True)
+
+    @property
+    def is_current(self):
+        return not self.wiki_page.is_deleted and int(self.identifier) == self.wiki_page.current_version_number
+
+    def html(self, node):
+        """The cleaned HTML of the page"""
+        sanitized_content = render_content(self.content, node=node)
+        try:
+            from bleach import linkify
+
+            return linkify(
+                sanitized_content,
+                [nofollow, ],
+            )
+        except TypeError:
```

Comment:
Is this being tested for here: https://github.com/CenterForOpenScience/osf.io/pull/8117/files#diff-85a582ce6c314b1ae17e90b485796d73R734, if so it should catch the log warning. I wish there was a little more specificity here, about what causes a TypeError especially because the wikis have sanitation on the client-side too and it's not always 100% clear where things are breaking when they break. But that's just an offhand suggestion.

Reply:
First, thanks for commenting on this method because it reminded me that Bleach was recently updated, and I needed to copy the recently changed html method from NodeWikiPage to WikiVersion.

Second, `test_bad_links` does not hit this TypeError.  I think it might be because of this bleach upgrade, but would need to verify.  Trying to think of content that would cause this TypeError to be hit in the first place(which linkify throws if something isn't a six.string_types.  non-strings seem like they would be caught before hitting the cleaning step.